### PR TITLE
Add Response code and errors in Protocol APIs

### PR DESCRIPTION
**Pull request by wchau**
Incremental changes from https://github.com/delta-io/delta-sharing/pull/85

Please review by viewing the markdown file directly.

https://github.com/delta-io/delta-sharing/blob/a701ef3b70af34a8e87ae6da93b71d6652f4e97d/PROTOCOL.md

### CODE REVIEW

File: PROTOCOL.md [hunk 1]
```diff
@@ -94,128 +240,815 @@ GET {prefix}/shares?maxResults=10&pageToken=...
...
+  "message": "string"
+}
+```
+
+</td>
+</tr>
+</table>
+</details>
+<details>
+<details>
```
Comment (linzhou-db) [inline PROTOCOL.md:579]: These `details` tag are not added properly If you check the preview, it folded everything below.

Comment (wchau) [reply PROTOCOL.md]: Whoops. Thanks!

**Comment by linzhou-db** [comment]
Are we going to cover 501/502/503/504? 
Are we going to cover other uncommon error codes?

**Comment by wchau** [comment]
> Are we going to cover 501/502/503/504? Are we going to cover other uncommon error codes?

No. I am only going to cover error codes that servers can typically control (along with its contents). This is consistent with databricks API https://docs.databricks.com/dev-tools/api/latest/account.html

**Review by linzhou-db — APPROVED** [review]

**Review by zhuansunxt — APPROVED** [review]
LGTM. I'm fine moving forward with this however I still prefer a dedicated error code section to avoid having scattered and duplicate text (there is no strong reason to comply with the format on Databricks documentation site).

**Comment by wchau** [comment]
> LGTM. I'm fine moving forward with this however I still prefer a dedicated error code section to avoid having scattered and duplicate text (there is no strong reason to comply with the format on Databricks documentation site).

Yeah, there is duplication here. If this turns out to be difficult to update, I will refactor it. Consistency is one of those small things that help others fall into the same gear.

**Pull request merged**